service/history/workflow/mutable_state_impl.go

```diff
@@ -649,6 +658,36 @@ func (ms *MutableStateImpl) GetQueryRegistry() QueryRegistry {
 	return ms.QueryRegistry
 }
 
+func (ms *MutableStateImpl) GetUpdateOutcome(
+	ctx context.Context,
+	updateID string,
+) (*updatepb.Outcome, error) {
+	ptr := ms.updateRecords[updateID].GetOutcomePointer()
+	if ptr == nil {
+		return nil, serviceerror.NewNotFound("update not found")
+	}
+	currentBranchToken, version, err := ms.getCurrentBranchTokenAndEventVersion(ptr.EventId)
```

Comment:
This call is cribbed from elsewhere but maybe these values should come from the pointer itself?